luarules/gadgets/unit_collision_damage_behavior.lua

```diff
@@ -147,25 +153,42 @@ end
 
 function gadget:GameFrame(frame)
 	for unitID, expirationFrame in pairs(unitInertiaCheckFlags) do
-		if not transportedUnits[unitID] then
+		if not transportedUnits[unitID] and not spGetUnitIsDead(unitID) then
 			local velX, velY, velZ, velocityLength = spGetUnitVelocity(unitID)
-			if velocityLength > velocityCap then --sheer off extreme velocities within acceptable range
-				velX = (velocityCap / velocityLength) * velX
-				velY = (velocityCap / velocityLength) * velY
-				velZ = (velocityCap / velocityLength) * velZ
+			
+			if velocityLength > velocityCap then -- Sheer off extreme velocities within acceptable range
+		
+				local horizontalVelocity = math.sqrt(mathAbs(velX)^2 + mathAbs(velZ)^2) --math abs to prevent NaN values
```

Comment:
hoes does abs prevent nan?

Reply:
```suggestion
				local horizontalVelocity = math.sqrt(velX^2 + velZ^2)
```
By definition x² is positive*, even if x is negative, so x²+z² can never yield a negative result.

* Unless x is complex, but we do not need to deal with complex numbers in BAR